### PR TITLE
Implement dynamic compatibility checks

### PR DESCRIPTION
**Pull request by DmitryTsepelev**
Dynamic checks for #52

### CODE REVIEW

File: spec/integrations/compatibility_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      it "returns ERROR when callback is passed" do
+        allow_any_instance_of(TestChannel).to receive(:follow) { |channel| channel.stream_from("all", -> {}) }
+        expect(subject.status).to eq :ERROR
+        expect(subject.error_msg).to include "Custom stream callbacks are not supported in AnyCable!"
+      end
+
+      it "returns ERROR when coder is passed" do
+        allow_any_instance_of(TestChannel).to receive(:follow) { |channel| channel.stream_from("all", coder: ActiveSupport::JSON) }
+        expect(subject.status).to eq :ERROR
+        expect(subject.error_msg).to include "Custom stream callbacks are not supported in AnyCable!"
```
Comment (palkan) [inline spec/integrations/compatibility_spec.rb]: I think, we should modify the message, smth like: "Custom coders are not supported".

Also, we can support `json` coder; it's the implicit Action Cable coder, so we don't have to raise an exception

Comment (DmitryTsepelev) [reply spec/integrations/compatibility_spec.rb]: Done, I guess in this case we want to use that valid coder somehow - right now our channel looks this way:

```ruby
module Channel
  class Base
    # ...

    def stream_from(broadcasting, _callback = nil, _options = {})
      connection.socket.subscribe identifier, broadcasting
    end

    # ...
  end
end
```

File: lib/anycable-rails.rb [hunk 2]
```diff
@@ -1,3 +1,7 @@
 # frozen_string_literal: true
 
 require "anycable/rails"
+
+module Anycable
+  class CompatibilityError < StandardError; end
```
Comment (palkan) [inline lib/anycable-rails.rb]: The error class shouldn't be defined here, we don't want to load the whole `anycable-rails` when loading compatibility tools.

Let's move it to `anycable/rails/compatibility.rb`

File: spec/integrations/compatibility_spec.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "spec_helper"
```
Comment (palkan) [inline spec/integrations/compatibility_spec.rb]: `"spec_helper"` loads `"anycable-rails"` but we want to test that `compatibility` works in isolation from any `anycable` code.

And we shouldn't test it in the context of RPC too.

So, the test setup should be:
- Load rails app with action cable
- Load `anycable/rails/compatibility`
- Call Action Cable code.

File: README.md [hunk 4]
```diff
@@ -147,6 +147,7 @@ Broadcasting | +
 Periodical Timers | -
 Disconnect remote clients | -
 
+Anycable Rails contains optional compatibility checks. In order to turn dynamic checks on you need to put `require "anycable/rails/compatibility"` into your `development` and `test` environments. After that `Anycable::CompatibilityError` would be thrown when your code tries to do use any of the incompatible features. Do not try to use it on production!
```
Comment (palkan) [inline README.md]: Let's remove this docs for now. There gonna be a separate repo for docs soon

Comment (DmitryTsepelev) [reply README.md]: done

File: lib/anycable-rails.rb (hunk 2 above)
Comment (DmitryTsepelev) [reply lib/anycable-rails.rb]: done

**Comment by palkan** [comment]
Merged through b71e4aa05ed2654f914d3c4173355f51f3cc0f53

**Pull request closed without merging**